dotnet-authserver/src/TeacherIdentity.AuthServer/Helpers/QueryStringSignatureHelper.cs

```diff
@@ -31,7 +31,9 @@ public bool VerifySignature(Url url)
     private string CalculateSignature(Url url)
     {
         var canonicalUrl = new Url(url.Path + "?" + url.Query)
-            .RemoveQueryParam(SignatureParameterName);
+            .RemoveQueryParam(SignatureParameterName)
+            .ToString()
+            .TrimEnd('?');
```

Comment:
There was some inconsistency previously when the `url` had no query parameters; sometimes this URL had a `?` at the end and sometimes it didn't (depending on exactly how it was constructed, seemingly).